framework/src/wifi_manager/wifi_manager.c

```diff
@@ -149,337 +318,270 @@ static wifi_utils_result_e stop_dhcp_server(void)
-	wifi_manager_scan_info_s *next_record;
 
-	curr_record = *wifi_manager_scan_info;
-	while (curr_record != NULL) {
-		next_record = curr_record->next;
-		free(curr_record);
-		curr_record = next_record;
+wifi_manager_result_e _wifimng_disconnect_ap(void)
+{
+	WM_LOG_START;
+	wifi_utils_result_e res = wifi_utils_disconnect_ap(NULL);
+	if (res != WIFI_UTILS_SUCCESS) {
+		ndbg("disconnect to ap fail\n");
+		return WIFI_MANAGER_FAIL;
 	}
-
-	*wifi_manager_scan_info = NULL;
+#ifdef CONFIG_ENABLE_IOTIVITY
+	if (result == WIFI_UTILS_SUCCESS) {
+		__tizenrt_manual_linkset("del");
+	}
+#endif
 	return WIFI_MANAGER_SUCCESS;
 }
 
-static wifi_manager_result_e wifi_fetch_scan_results(slsi_scan_info_t **wifi_scan_info, wifi_manager_scan_info_s **wifi_manager_scan_info)
+
+wifi_manager_result_e _wifimng_run_softap(wifi_manager_softap_config_s *config)
 {
-	wifi_manager_scan_info_s *curr_record;
-	wifi_manager_scan_info_s *prev_record = NULL;
-	slsi_scan_info_t *wifi_scan_iter = NULL;
-	int i = 0;
-	if (*wifi_scan_info == NULL) {
+	WM_LOG_START;
+	wifi_utils_softap_config_s softap_config;
+
+	softap_config.channel = config->channel;
+	softap_config.ap_crypto_type = WIFI_UTILS_CRYPTO_AES;
+	softap_config.ap_auth_type = WIFI_UTILS_AUTH_WPA2_PSK;
+	softap_config.ssid_length = strlen(config->ssid);
+	softap_config.passphrase_length = strlen(config->passphrase);
+	strncpy(softap_config.ssid, config->ssid, softap_config.ssid_length + 1);
+	strncpy(softap_config.passphrase, config->passphrase, softap_config.passphrase_length + 1);
+
+	wifi_utils_result_e res = wifi_utils_start_softap(&softap_config);
+	if (res != WIFI_UTILS_SUCCESS) {
+		ndbg("Starting softap mode failed.");
 		return WIFI_MANAGER_FAIL;
 	}
-	/* Initialize pointer */
-	wifi_scan_iter = *wifi_scan_info;
 
-	do {
-		curr_record = (wifi_manager_scan_info_s *)malloc(sizeof(wifi_manager_scan_info_s));
-		if (curr_record == NULL) {
-			wifi_free_scan_results(wifi_manager_scan_info);
-			return WIFI_MANAGER_FAIL;
-		}
-		if (!prev_record) {
-			*wifi_manager_scan_info = curr_record;
-		} else {
-			prev_record->next = curr_record;
-		}
+	if (_start_dhcpd() != WIFI_MANAGER_SUCCESS) {
+		ndbg("Starting DHCP server failed.\n");
+		return WIFI_MANAGER_FAIL;
+	}
 
-		memset(curr_record->ssid, 0x00, SLSI_SSID_LEN + 1);
-		memset(curr_record->bssid, 0x00, SLSI_MACADDR_STR_LEN);
-		curr_record->rssi = wifi_scan_iter->rssi;
-		curr_record->channel = wifi_scan_iter->channel;
-		curr_record->phy_mode = wifi_scan_iter->phy_mode;
-		strncpy(curr_record->ssid, (char *)wifi_scan_iter->ssid, SLSI_SSID_LEN + 1);
-		strncpy(curr_record->bssid, (char *)wifi_scan_iter->bssid, SLSI_MACADDR_STR_LEN);
-		prev_record = curr_record;
-		wifi_scan_iter = wifi_scan_iter->next;
-		i++;
-	} while (wifi_scan_iter != NULL);
-
-	curr_record->next = NULL;
-	printf("%d records scanned\n", i);
+	/* update wifi_manager_info */
+	LOCK_WIFIMNG_INFO;
+	strncpy(g_manager_info.ssid, config->ssid, softap_config.ssid_length + 1);
```

Comment:
Does  connection status need an update here?

Reply:
I think so. because it change mode. so SSID should be changed.